storage/storage_dest.go

```diff
@@ -84,18 +84,25 @@ type storageImageDestinationLockProtected struct {
 	currentIndex          int                    // The index of the layer to be committed (i.e., lower indices have already been committed)
 	indexToAddedLayerInfo map[int]addedLayerInfo // Mapping from layer (by index) to blob to add to the image
 
-	// In general, a layer is identified either by (compressed) digest, or by TOC digest.
+	// In general, a layer is identified either by (compressed) digest, or by TOC digest
+	// (and we assume the TOC digest also uniquely identifies the contents, i.e. there aren’t two
+	// different formats/ways to parse a single TOC).
```

Comment:
All of the c/storage+c/image code has been built around this assumption, but it is false currently (https://github.com/containers/storage/issues/1888 ) and I’m not sure whether we need to revisit the design. Let’s discuss that in the c/storage issue.

Reply:
yes, this assumption is correct